service.toml

```diff
@@ -20,30 +20,92 @@ implement = ["multiparter"]
 
 [namespace.storage.new]
 required = ["location", "name"]
-optional = ["default_storage_pairs", "pair_policy", "work_dir"]
+optional = ["default_storage_pairs", "pair_policy", "work_dir", "excepted_bucket_owner"]
 
 [namespace.storage.op.delete]
-optional = ["multipart_id"]
+optional = ["multipart_id", "excepted_bucket_owner"]
 
 [namespace.storage.op.list]
 optional = ["list_mode"]
 
 [namespace.storage.op.read]
-optional = ["offset", "io_callback", "size"]
+optional = ["offset", "io_callback", "size", "excepted_bucket_owner", "sse_customer_algorithm", "sse_customer_key", "sse_customer_key_md5"]
 
 [namespace.storage.op.write]
-optional = ["content_md5", "content_type", "io_callback", "storage_class"]
+optional = ["content_md5", "content_type", "io_callback", "storage_class", "excepted_bucket_owner", "bucket_key_enabled", "sse_customer_algorithm", "sse_customer_key", "sse_customer_key_md5", "sse_kms_key_id", "server_side_encryption"]
+
+[namespace.storage.op.stat]
+optional = ["excepted_bucket_owner", "sse_customer_algorithm", "sse_customer_key", "sse_customer_key_md5"]
+
+[namespace.storage.op.create_multipart]
+optional = ["bucket_key_enabled", "excepted_bucket_owner", "sse_customer_algorithm", "sse_customer_key", "sse_customer_key_md5", "sse_kms_key_id", "server_side_encryption"]
+
+[namespace.storage.op.write_multipart]
+optional = ["excepted_bucket_owner", "sse_customer_algorithm", "sse_customer_key", "sse_customer_key_md5"]
+
+[namespace.storage.op.complete_multipart]
+optional = ["excepted_bucket_owner"]
 
 [pairs.default_service_pairs]
 type = "DefaultServicePairs"
 description = "set default pairs for service actions"
 
+[pairs.force_path_style]
+type = "bool"
+description = "see http://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html for Amazon S3: Virtual Hosting of Buckets"
+
+[pairs.disable_100_continue]
+type = "bool"
+description = "set this to `true` to disable the SDK adding the `Expect: 100-Continue` header to PUT requests over 2MB of content"
+
+[pairs.use_accelerate]
+type = "bool"
+description = "set this to `true` to enable S3 Accelerate feature"
+
+[pairs.disable_content_md5_validation]
```

Comment:
Remove this pair as we always disabled it to allow users to calculate them manully.